Tchap/Utils/Tools.m

```diff
@@ -150,10 +150,43 @@ + (NSString *)permalinkToRoom:(NSString *)roomIdOrAlias
     return [NSString stringWithFormat:@"%@/#/room/%@", urlPrefix, roomIdOrAlias];
 }
 
++ (NSString *)permalinkToRoomWithoutAliasFromRoomState:(MXRoomState *)state {
+    NSString *roomId = state.roomId;
+    
+    NSArray<MXRoomMember *> *members = state.members.joinedMembers;
+    
+    NSMutableSet<NSString *> *memberHomeservers = [NSMutableSet setWithCapacity:members.count];
+    
+    // Add the homeServer hosting the room.
+    [memberHomeservers addObject:[roomId componentsSeparatedByString:@":"].lastObject];
+    
+    // List unique members' homeServers in memberHomeservers. Limit to 3
+    [members enumerateObjectsUsingBlock:^(MXRoomMember * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+        NSArray<NSString *> *userIdParts = [obj.userId componentsSeparatedByString:@":"];
+        if (userIdParts.count > 1)
+        {
+            [memberHomeservers addObject:userIdParts.lastObject]; // NSMutableSet only add object if not already present in the Set.
+            
+            // Stop if 3 homeServers are listed.
+            if (memberHomeservers.count >= 3)
+            {
+                *stop = YES;
+            }
+        }
+    }];
+    
+    NSString *urlPrefix = BuildSettings.clientPermalinkBaseUrl;
+    NSString *viaParameters = [NSString stringWithFormat:@"?via=%@", [memberHomeservers.allObjects componentsJoinedByString:@"&via="]];
+    NSString *permalinkToRoom = [NSString stringWithFormat:@"%@/#/room/%@%@", urlPrefix, roomId, viaParameters];
```

Comment:
You should update `permalinkToRoom` in `Tools.m` from Tchap > Utils and using it. What do you think about that ?

Reply:
I thought about it be didn't make it because the modification here, building `via` parameters does a lot more stuff than `permalinkToRoom`. It will denaturate `permalinkToRoom`.
That's why I prefered to define a standalone `permalinkToRoomWithoutAliasFromRoomState`.